client/src/extension.ts

```diff
@@ -122,21 +122,20 @@ export function activate(context: ExtensionContext) {
     },
   };
 
-  // Options to control the language client
-  let clientOptions: LanguageClientOptions = {

```

Comment:
Is there a way to make sure `let clientOptions` is run instead? Why are we not doing exactly the same steps as the first start, on a restart?
Are there any steps that should *not* be repeated?

Reply:
E.g. at the coarsest level why not deactivate the client and activate it again.

Reply:
I don't quite follow what you mean. `let clientOptions` is run...?

Reply:
> I don't quite follow what you mean. `let clientOptions` is run...?

The "function containing it" is run.

Reply:
Or else, we need to ask the same question of any other `let foo` in that file, which upon future changes, might not do what is's expected to do after a restart.

Reply:
Also a curiosity: who calls `deactivate` and when?

Reply:
@cristianoc done.